meson.build

```diff
@@ -98,6 +98,7 @@ feature_map = {
   'audit-events'                                : '-DBMCWEB_ENABLE_LINUX_AUDIT_EVENTS',
   #'vm-nbdproxy'                                : '-DBMCWEB_ENABLE_VM_NBDPROXY',
   'ibm-led-extensions'                          : '-DBMCWEB_ENABLE_IBM_LED_EXTENSIONS',
+  'hw-isolation'                                : '-DBMCWEB_ENABLE_HW_ISOLATION',
```

Comment:
As we experienced the compilation error by not enabling `hw-isolation`, I think we can start to use  the compile time evalution like
```
if constexpr (bmcwebHwIsolationEnabled) {
....
```

For it. update `meson.build` and `bmcweb_config.in.h` similar to `BMCWEB_ENABLE_HEALTH_POPULATE`.

meson.build:
```
enable_hw_isolation = get_option('hw-isolation')
conf_data.set10('BMCWEB_ENABLE_HW_ISOLATION', enable_hw_isolation.allowed())
```

bmcweb_config.h.in:

```
constexpr const bool bmcwebHwIsolationEnabled = @BMCWEB_ENABLE_HW_ISOLATION@ == 1;
```

Then, the places where we use `#ifdef BMCWEB_ENABLE_HW_ISOLATION` can be changed like

```
#include <bmcweb_config.h>

...
if constexpr(bmcwebHwIsolationEnabled)
{
...
}
```

This way, all of parts (even under bmcwebHwIsolationEnabled) will go thru the compilation even if the part (if it is not enabled) may not be emitted to binary at the end.
This will prevent the compilation mistake which is caused by the forgotten enablement.